### ISSUE TITLE
allow transport options in node.js to allow connection via a proxy

### ISSUE THREAD

**Issue by colmaengus** [crossbario/autobahn-js#204]
Hi,

I need to use autobahn in node.js behind a proxy.
To do this I would normally pass an agent option to the websocket creation.
However I don't have a way to do this in autobahn.
With a small change this could be added as an transport option
e.g.

`this.connection = new autobahn.Connection({
    transports: [
      {
        type: 'websocket',
        url: router,
        agent: agent
      }
    ],
    realm: realm
  });`

The lib/transport/websocket.js would then require a small change to pass this to websocket.

What do you think ?

Regards

Colm A

**Comment by colmaengus** [comment]
actually it would be useful to be able to provide quite a few of the options that websocket supports e.g. cert, key (for client certs when using e.g. nginx to provide the https connection)

**Comment by colmaengus** [comment]
I see this issue has already come up before (#163), none the less a fix for running node.js behind a proxy would be most welcome.

**Comment by oberstet** [comment]
@colmaengus yes, I definitely see the need. for this, an for more things like #163. The main question is about API: a sane, clean way of exposing knobs without (probably) exposing too much implementation details (the use of "ws" as the underlying WebSocket library when running on Node)

**Comment by colmaengus** [comment]
Adding support for an transport agent would be a difference between node.js and browser but its common to allow an agent to be specified in node.js. I've tried replacing the globalAgent in node before requiring autobahn but its not working.
